### DISCUSSION TITLE
Tests infrastructure for composition

### DISCUSSION THREAD

**Discussion by rdeioris** [syoyo/tinyusdz#211] [Ideas]
Hi, right now the  tests/usda/composition directory is not honoured by tests, and the current approach is to just test the "parsability" of the files, while for composition we probably want to test the "flattened result".

Would you accept a patch for an additional test runner that given composition input files runs tusdcat and compare the flattened result?

**Comment by syoyo** [comment]
> Would you accept a patch for an additional test runner that given composition input files runs tusdcat and compare the flattened result?

Yes, its appreciated! I was thinking about it. The idea in my mind is using Python difflib https://docs.python.org/3/library/difflib.html or FileCheck https://llvm.org/docs/CommandGuide/FileCheck.html to compare the USD composition result with (manually tailored)  the flattened result.

What test runner for USD composition are you planning in your patch?
